hera_cal/lstbin.py

```diff
@@ -190,13 +222,16 @@ def lst_bin(data_list, lst_list, flags_list=None, dlst=None, begin_lst=None, lst
     else:
         # keep all lst_grid bins and fill empty ones with unity data and mark as flagged
         for index in range(len(lst_grid)):
-            if index in all_lst_indices:
-                # skip if index already in data
-                continue
+            # I got rid of this statement because it causes failures if a
```

Comment:
Once this PR is going to be pulled in, this will be a very confusing comment!

While I think I'm fine taking out this check, I'm wondering why it was here... Feels like a Chesterton's Fence...

Reply:
I think the reason it was here was that there was no reason to fill in fake data because this lst had been visited by a previous night and if we assume that all nights have the same baselines, then this baseline has already been spoofed. If we let baselines vary between nights, this assumption is no longer true so we can't necessarily skip spoofing.

Reply:
To keep the check, and whatever performance speedups where intended, i've added another check that checks whether baselines are the same across all nights. If they are, then we can keep this check and skip the following spoofing for loop.